### PR TITLE
#42597 Release notes: Conditional Form Flow steps

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: using-valtimo/form-flow/create-form-flow-definition.md [hunk 1]
```diff
@@ -17,119 +17,157 @@ To create form flow definition, the following steps are necessary:
...
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
+
+* The `personalDetailsStep` is now followed by two other steps. This is only allowed when at least one of the two steps
```
Comment (theo-ritense) [inline using-valtimo/form-flow/create-form-flow-definition.md:74]: If it is required to have at least one conditional step, please don't given an example (above) that is invalid.

File: using-valtimo/form-flow/create-form-flow-definition.md [hunk 2]
```diff
@@ -17,119 +17,157 @@ To create form flow definition, the following steps are necessary:
...
+            "nextStep": "summaryStep"
+        },
+        {
+            "key": "summaryStep"
+        }
+    ]
+}
+```
 
 * Configure the step type. Currently, the only step type supported is `form`, which requires a `definition` property to
```
Comment (theo-ritense) [inline using-valtimo/form-flow/create-form-flow-definition.md:111]: Same. If type is required, don't show the example without it. You might get away with adding
```
"type": { ... }
```
in the example above.

File: using-valtimo/form-flow/create-form-flow-definition.md [hunk 3]
```diff
@@ -17,119 +17,157 @@ To create form flow definition, the following steps are necessary:
...
             }
-        ]
-    }
-    ```
+        }
+    ]
+}
+```
 
 * Add triggers to steps where necessary (e.g. to store data externally). See below for more information.
```
Comment (theo-ritense) [inline using-valtimo/form-flow/create-form-flow-definition.md:172]: Please state the names of the triggers 'onOpen', 'onBack' that the user should look for in the example. Now it's a guessing game which 'triggers' you are referring to.

File: using-valtimo/form-flow/create-form-flow-definition.md (hunk 1 above)
Comment (Klaas-Ritense) [reply using-valtimo/form-flow/create-form-flow-definition.md:74]: The first few examples also miss a `type` which makes them invalid. It section just gives a step by step approach to build a valid form flow

File: using-valtimo/form-flow/create-form-flow-definition.md (hunk 3 above)
Comment (Klaas-Ritense) [reply using-valtimo/form-flow/create-form-flow-definition.md:172]: The information below states what the triggers are. Its collapsed

File: using-valtimo/form-flow/create-form-flow-definition.md (hunk 1 above)
Comment (marijnritense) [reply using-valtimo/form-flow/create-form-flow-definition.md:74]: I think it's okay to keep to this approach. This page is meant to be followed step by step, so it makes sense to only introduce the elements you're talking about at that point in time.

File: using-valtimo/form-flow/create-form-flow-definition.md (hunk 2 above)
Comment (marijnritense) [reply using-valtimo/form-flow/create-form-flow-definition.md:111]: Type is introduced in this section as to not overwhelm users. I think it's fine to keep it as-is.

File: release-notes/major10/10.1.0/valtimo-backend-libraries.md [hunk 4]
```diff
@@ -20,6 +20,11 @@ The following features were added:
   the id of a form flow instance. More
   information can be found [here](/reference/modules/form-flow.md#available-properties-in-spel-context)
 
+* **Form flow conditional next step**
+
+  A next step in a form flow can now be conditional. This allows for more complex form flows. More information can be
+  found [here](/using-valtimo/form-flow/create-form-flow-definition.md)
```
Comment (marijnritense) [inline release-notes/major10/10.1.0/valtimo-backend-libraries.md:26]: Maybe also mention the submission data type change here again with a reason as to why the change is made (e.g. to make it possible to directly access fields of the submission data in expressions)?

**Review by marijnritense — APPROVED** [review]
